src/psp.ts

```diff
@@ -0,0 +1,664 @@
+                airnodeAddress,
+                oises,
+                apiCredentials,
+                id: subscriptionId,
+                templateId,
+                overrideParameters,
+                oisTitle,
+                endpointName,
+                endpointId: template.endpointId,
+                templateParameters: template.templateParameters,
+              })
+            );
+            if (!isNil(errorCallApi) || isNil(logsData)) {
+              node.logger.warn('Failed to fecth API value', subscriptionIdLogOptions);
+              continue;
+            }
+            const [logs, data] = logsData;
+            node.logger.logPending(logs, subscriptionIdLogOptions);
+
+            const apiValue = data[subscriptionId];
+            if (isNil(apiValue)) {
+              node.logger.warn('API value not found. Skipping update...', subscriptionIdLogOptions);
+              continue;
+            }
+
+            // **************************************************************************
+            // Check conditions
+            // **************************************************************************
+            node.logger.debug('Checking conditions...', subscriptionIdLogOptions);
+
+            const encodedFulfillmentData = ethers.utils.defaultAbiCoder.encode(['int256'], [apiValue]);
+            let conditionFunction: ethers.utils.FunctionFragment;
+            let conditionParameters: string;
+            try {
+              const decodedConditions = abi.decode(conditions);
+              const [decodedConditionFunctionId] = ethers.utils.defaultAbiCoder.decode(
+                ['bytes4'],
+                decodedConditions._conditionFunctionId
+              );
+              conditionFunction = dapiServer.interface.getFunction(decodedConditionFunctionId);
+              conditionParameters = decodedConditions._conditionParameters;
+            } catch (err) {
+              node.logger.error('Failed to decode conditions', {
+                ...subscriptionIdLogOptions,
+                error: err as any,
+              });
+              continue;
+            }
+            const [errorConditionFunction, [conditionsMet]] = await retryGo(() =>
+              dapiServer
+                .connect(voidSigner)
+                .functions[conditionFunction.name](subscriptionId, encodedFulfillmentData, conditionParameters)
+            );
+            if (errorConditionFunction) {
+              node.logger.error('Failed to check conditions', {
+                ...subscriptionIdLogOptions,
+                error: errorConditionFunction,
+              });
+              continue;
+            }
+            if (!conditionsMet) {
+              node.logger.warn('Conditions not met. Skipping update...', subscriptionIdLogOptions);
+              continue;
+            }
+
+            // **************************************************************************
+            // Fetch current gas fee data
```

Comment:
We probably can get away with doing this only once while getting the block number